src/core/load_balancing/lb_policy.h

```diff
@@ -116,27 +118,33 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
    public:
     virtual ~MetadataInterface() = default;
 
-    //////////////////////////////////////////////////////////////////////////
-    // TODO(ctiller): DO NOT MAKE THIS A PUBLIC API YET
-    // This needs some API design to ensure we can add/remove/replace metadata
-    // keys... we're deliberately not doing so to save some time whilst
-    // cleaning up the internal metadata representation, but we should add
-    // something back before making this a public API.
-    //////////////////////////////////////////////////////////////////////////
-
-    /// Adds a key/value pair.
-    /// Does NOT take ownership of \a key or \a value.
-    /// Implementations must ensure that the key and value remain alive
-    /// until the call ends.  If desired, they may be allocated via
-    /// CallState::Alloc().
-    virtual void Add(absl::string_view key, absl::string_view value) = 0;
+    virtual absl::optional<absl::string_view> Lookup(
+        absl::string_view key, std::string* buffer) const = 0;
 
     /// Produce a vector of metadata key/value strings for tests.
```

Comment:
Noting it would be nice to figure out how to eliminate this

Reply:
Good point.  And this PR actually made it easy to do that, so I've gone ahead and done so.

Reply:
TestOnlyCopyToVector still seems to be here

Reply:
It's gone from the public API, only present in the internal implementation, which the tests reach into.